stlab/concurrency/channel.hpp

```diff
@@ -930,20 +930,17 @@ struct shared_process
             } else {
                 if (get_process_state(_process).first == process_state::await) return;
             }
-            auto now = std::chrono::steady_clock::now();
-            process_state state;
-            std::chrono::steady_clock::time_point when;
-            std::tie(state, when) = get_process_state(_process);
+            auto [state, duration] = get_process_state(_process);
```

Comment:
I love structured bindings but wont this require c++17 rather than just c++14?

Reply:
@FelixPetriconi I was really concerned about this change. I am and avid user of channels and currently have a hard requirement of c++14 in my embedded application. I would understand if you guys wanted to require c++17 or later but if that is the case there are several other changes I would suggest like using `weak_from_this()` rather than `make_weak_ptr()`.

Reply:
We already used constexpr if in the code which is a C++17 feature as well. So I didn't thought that this is a problem. But I can add a workaround.

Reply:
After you mentioned that we are using `if constexpr` I was curious how our project was even compiling.

Turns out that even though we are compiling with `-std=c++14` the XCode version of clang is simply emitting `warning: constexpr if is a C++17 extension [-Wc++17-extensions]`.

Our project runs on a linux distro that only has gcc 6.3 and `if constexpr` was added in 7 so that got me concerned. However I don't think it is a problem since `if constexpr` is evaluated at compile time and the invalid branches are always removed if it is used within a template. Therefore there is not new code that it would have to reference in gcc.

I think the same will be true for structured bindings since it is just syntactic sugar rather than any actual new logic. You may not need to add a work around for my case.